### ISSUE TITLE
Calling database on passing IQueryable to Serilog

### ISSUE THREAD

**Issue by st249** [serilog/serilog#2152]
**Description**
When i pass an IQueryable to log, it translates and calls database

**Reproduction**
This is a sample code, i getAll the repository and when pass it as parameter to serilog it translates and get all records from table, this is leading to ram leakage in our application. We are not using the serilog like this but somehow in some exceptions there are queryable items which are creating unpredictable behavior
  var repo = _genericUnitOfWork.GetRepository<DistributionLocation>();
  var x = repo.GetAll();
  _logger.LogWarning("ssss {@x}", x);

**Expected behavior**
Do not call any query on database because the user doesn't expect to call a query in setting logs.

**Relevant package, tooling and runtime versions**
		<PackageReference Include="Serilog.AspNetCore" Version="3.2.0" />
		<PackageReference Include="Serilog.Exceptions" Version="5.4.0" />
		<PackageReference Include="Serilog.Settings.Configuration" Version="3.1.0" />
		<PackageReference Include="Serilog.Sinks.Console" Version="3.1.1" />
		<PackageReference Include="Serilog.Sinks.Http" Version="5.2.1" />
		<PackageReference Include="Serilog.Sinks.Seq" Version="4.0.0" />
		<PackageReference Include="serilog.sinks.graylog" Version="2.1.3" />
<PackageReference Include="Serilog.Enrichers.Environment" Version="2.1.3" />

**Comment by nblumhardt** [comment]
Hi! The `@` in:

```csharp
_logger.LogWarning("ssss {@x}", x);
```

means "serialize". Many types aren't safe to serialize, hence this is an opt-in. It sounds like in your scenario, you don't want to serialize `x`, since doing so will access properties that result in database accesses.

Changing the code to just:

```csharp
_logger.LogWarning("ssss {x}", x);
```

is the intended usage. HTH!

**Comment by st249** [comment]
No, i just wanted to clear my purpose. This is not my code, it is a sample case to show that serilog is calling **DATABASE** when a **Quaryable** is passing to it. 
In our scenario, we had a ram issue and we reached there that somehow a query without Where clause is running in application and this lead to fetching all data to RAM. The stacktrace log showed us a line in which we were logging the whole exception. I got suspicious of serilog, and then tested with a dev endpoint the code i provided you and i found that serilog is calling database and getting the data from database.

Everything clear?

**Comment by nblumhardt** [comment]
Hi! Yes, this is clear, but the cause is likely to be the same - whatever code in your application is calling Serilog in this situation, it's requesting that the queryable object be serialized. If you need more info on this, it's best to publish a complete example to Stack Overflow with the `serilog` tag. Cheers!

**Comment by st249** [comment]
So you say it is ok that the queryable serialization logging is calling database??? Isn't it an issue ?

**Comment by Numpsy** [comment]
fwiw there are some notes about this sort of situation over in [Serilog.Exceptions](https://github.com/RehanSaeed/Serilog.Exceptions?tab=readme-ov-file#serilogexceptionsentityframeworkcore).
I suppose in the general case, Serilog can't know if something is going to call a database when accessed though.

**Comment by st249** [comment]
I think it is in the process of serialization. it seems that a ToList() function on queryable is ran in serialization process, so if this process of serializing the log is not in the serilog core, i will not bother you anymore. But if serialization is in the core of serilog i think there is an issue.

**Comment by sungam3r** [comment]
@st249 The general rule is simple - you should not pass arbitrary objects into logs - Serilog, MEL or whatever. As @nblumhardt said
>  Many types aren't safe to serialize

Imagine you pass an instance of `System.Windows.Form` class with `@` qualifier into log . Experience may vary but there will be nothing good.

**Comment by st249** [comment]
we are not doing that , we are logging an exception and this behavior has been happened.

**Comment by sungam3r** [comment]
Then I suggest you to add destructuring policy that filters out any property of `IQueryable` type.

**Comment by st249** [comment]
ok thanks

**Comment by nblumhardt** [comment]
> we are logging an exception and this behavior has been happened

The Serilog.Exceptions link above will have the info you need, this is most commonly a misconfiguration of that package.

**Comment by sungam3r** [comment]
One suggestion - make static field in that pilicy and return it instead of newing.
